File: app/src/main/java/com/qzb/scoketlink/MyApplication.java
package com.qzb.scoketlink;

import android.app.Application;
import android.net.nsd.NsdManager;
import android.provider.Settings;

public class MyApplication extends Application {

    private NsdManager nsdManager;


    @Override
    public void onCreate() {
        super.onCreate();
        NsdHelper nsdHelper = NsdHelper.get(this);
        nsdManager = nsdHelper.init(NsdHelper.getSelfServiceName(), 10001);
    }

    public NsdManager getMsdManager() {
        return nsdManager;
    }

}

File: smaple/src/main/java/com/qzb/smaple/MyApplication.kt
package com.qzb.smaple

import android.app.Application
import android.widget.Toast
import com.qzb.socket.ServerListener
import com.qzb.socket.SocketUtils
import org.java_websocket.WebSocket

/**
 * 生活白科技-居家小确幸
 *
 * @ClassName: MyApplication
 * @Author: Leon.Qin
 * @Date: 2022/8/24 14:04
 * @Description:
 */
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        SocketUtils.init(this)
    }
}